docs/jobs_usage.rst

```diff
@@ -1,23 +1,172 @@
+
+   while True:
+       job = await client.jobs.status(job.id)
+       if job.status in (JobStatus.RUNNING, JobStatus.SUCCEEDED):
+           break
+       elif job.status == JobStatus.FAILED:
+           raise RuntimeError(f"Job {job.id} failed with {job.reason}:"
+                              f"{job.history.description}")
+       else:
+           await asyncio.sleep(1)
+
+
+Mount Neuromation Storage folders
+=================================
+
+The Neuromation Platform provides access to Neuromation storage (``storage://``) by
+*mounted folders* inside a container (*volumes* in `Docker <https://www.docker.com>`_
+glossary).
+
+
+If you have a directory ``storage:folder`` and want to mount it inside a container under
+``/var/data`` path please create a :class:`Volume` and use it in :class:`Container`
+definition::
+
+   from yarl import URL
+
+   volume = Volume(
+       storage_uri=URL("storage:folder"),
+       container_path="/var/data",
+   )
+
+   job = await client.jobs.run(
+       Container(
+           image=client.parse.remote_image("ubuntu:latest"),
+           resources=Resources(memory_mb=100, cpu=0.5),
+           command="sleep 30",
+           volumes=[volume],
+       )
+   )
+
+
+There is a parsing helper that can construct a :class:`Volume` instance from a string in
+format that is used in :term:`CLI`::
+
+   volume = client.parse.volume("storage:folder:/var/data")
+
+.. seealso:: :ref:`storage-usage` for the storage manipulation API.
+
+
+Kill a Job
+==========
+
+Use :meth:`Jobs.kill` for enforcing job to stop::
+
+   await client.jobs.kill(job.id)
 
 
+Expose job's TCP ports locally
+==============================
+
+Sometimes there is a need to access TCP endpoints exposed by a job executed on the
+Neuromation Platform from local workstation.
+
+For example, you've started a gRPC server inside a container on TCP port ``12345`` and
+want to access this service from your laptop.
+
+You need to bridge this *remote* ``12345`` port into a local TCP namespace
+(e.g. ``23456`` *local* port by :meth:`Jobs.port_forward` method::
+
+   from grpclib.client import Channel
+
+   # generated by protoc
+   from .helloworld_pb2 import HelloRequest, HelloReply
+   from .helloworld_grpc import GreeterStub
+
+    async with client.jobs.port_forward(job.id, 23456, 12345):
+        # open gRPC client and use it
+
+        channel = Channel('127.0.0.1', 50051)
+        greeter = GreeterStub(channel)
+
+        reply: HelloReply = await greeter.SayHello(HelloRequest(name='Dr. Strange'))
+        print(reply.message)
+
+        channel.close()
+
+The example uses `grpclib <https://github.com/vmagamedov/grpclib>`_ library for make
+client gRPC requests.
+
 .. _job-preemption:
 
 Job preemption
 ==============
 
-TBD
+Job preemption means that unlike normal jobs preemptible ones can be stopped by kernel
+when the system has lack of resources and restarted later.  All memory changes are lost
+but disk data is persistent.
```

Comment:
should we point out that by persistent disk here we mean specifically the Neuromation Storage mounts? the host disks are lost.

Reply:
the doc is updated